sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPump.java

```diff
@@ -61,14 +68,25 @@ void setLastEnqueuedEventProperties(LastEnqueuedEventProperties lastEnqueuedEven
      */
     @Override
     public void close() {
+        // When client receive complete signal will call this method again in clean up method on scheduler thread,
+        // the first dispose will cause a timeout exception. Add a check avoid close twice.
+        if (isDisposed.getAndSet(true)) {
+            return;
+        }
         try {
             client.close();
         } catch (Exception error) {
             LOGGER.atInfo()
                 .addKeyValue(PARTITION_ID_KEY, partitionId)
                 .log("Exception occurred disposing of consumer client.", error);
         } finally {
-            scheduler.dispose();
+            try {
```

Comment:
I was wondering if there's any way to restructure this. Do we need a nested try/catch? 

I thought it wasn't possible to have a try/catch inside a finally block... but I guess it works.

Reply:
Yeah, it looks not good. I saw there are some try/catch inside a finally block in our library, not sure our code perference, it's better to put this try/catch outside finally block?